src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/Model/Field.cs

```diff
@@ -5,7 +5,27 @@
 
 namespace CSharpSyntaxGenerator
 {
-    public class Field
+    public class TreeTypeChild
+    {
+    }
+
+    public class Choice : TreeTypeChild
+    {
+        [XmlElement(ElementName = "Field", Type = typeof(Field))]
+        [XmlElement(ElementName = "Choice", Type = typeof(Choice))]
```

Comment:
>Choice [](start = 35, length = 6)

Does it make sense to have a choice nested in a choice?

Reply:
yes!  It actually happens :)    For example, with properties you have: 

```
| attribute_list* modifier* type explicit_interface_specifier? identifier_token (accessor_list | ((arrow_expression_clause | equals_value_clause) ';'))
```

Reply:
oh... i see waht you mean.  in this case, we have a `sequence` that is the child of a `choice`.    And a `choice` that is the child of that `sequence`.   I will change ths up!  thanks!

Reply:
Resolved. Thanks!